cli/command/secret/create.go

```diff
@@ -46,28 +50,26 @@ func runSecretCreate(dockerCli command.Cli, options createOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	var in io.Reader = dockerCli.In()
-	if options.file != "-" {
-		file, err := system.OpenSequential(options.file)
-		if err != nil {
-			return err
-		}
-		in = file
-		defer file.Close()
+	if options.driver != "" && options.file != "" {
+		return errors.Errorf("When using secret driver secret data must be empty")
```

Comment:
@thaJeztah according to the secrets plugin subsystem [design](https://docs.google.com/document/d/1kNA6apIFAiaE9d96A2_rcEulVrglvq05zEXGdijSZac), the secrets plugin will be readonly (that is, swarm will not populate the secret values). 
I think it might be better to prevent incorrect usage as soon as possible (since `file` is a cli only option).